bin/server/cmd/component_slim.go

```diff
@@ -0,0 +1,37 @@
+//go:build slim
+// +build slim
+
+package cmd
+
+import (
+	"fmt"
+
+	"github.com/bytebase/bytebase/store"
+	"go.uber.org/zap"
+)
+
+type MetadataDBManager struct {
+	profile *Profile
+	l       *zap.Logger
+}
+
+func createMetadataDBManager(profile *Profile, logger *zap.Logger) (*MetadataDBManager, error) {
+	if useEmbeddedDB() {
+		return nil, fmt.Errorf("slim build requires using --pg to specify an external PostgreSQL instance ")
+	}
+	return &MetadataDBManager{
+		profile: profile,
+		l:       logger,
+	}, nil
+}
+
+func (m *MetadataDBManager) newDB() (*store.DB, error) {
+	if useEmbeddedDB() {
+		return nil, fmt.Errorf("slim build doesn't embed the PostgreSQL binary. Please either use --pg to specify an external PostgreSQL instance or use the full build embedding the PostgreSQL binary")
```

Comment:
```suggestion
		return nil, fmt.Errorf("slim build doesn't embed the PostgreSQL binary. Please either use --pg to specify an external PostgreSQL instance or use the full build embedding the PostgreSQL binary.")
```

Reply:
I guess we don't need if block here since createMetadataDBManager has already checked the logic

Reply:
It's about defensive programming but it's OK to remove this block if you insist it.

Reply:
Oh, let's remove the "." then

Reply:
> It's about defensive programming but it's OK to remove this block if you insist it.

I did think a bit here. another option is to return nil, fmt.Errorf("should not reach here") in both newDB() and close().

Reply:
@tianzhou You convince me to remove the if block. However, the `close` function should never return error on this case because the expected close behavior of `MetadataDBManager` in `slim` mode is empty.

Reply:
Updated.